yamux/src/session.rs

```diff
@@ -605,10 +595,17 @@ where
             return Poll::Ready(Some(Err(io::ErrorKind::TimedOut.into())));
         }
 
-        self.recv_frames(cx)?;
-        self.recv_events(cx)?;
+        loop {
+            if self.is_dead() {
+                break;
+            }
 
-        self.flush(cx)?;
+            let mut is_pending = self.recv_frames(cx)?.is_pending();
+            is_pending &= self.recv_events(cx)?.is_pending();
+            if is_pending {
+                break;
+            }
+        }
```

Comment:
Why not flush after `loop {}`?

Reply:
if pending，flush is redundant operations，if ready, next poll will flush